agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh

```diff
@@ -222,20 +225,47 @@ function install_k8s_agent {
         echo -e "        enabled: true" >> $CONFIG_FILE
     fi
 
-    sed -i -e "s|# serviceAccount: sysdig-agent|serviceAccount: sysdig-agent|" /tmp/sysdig-agent-daemonset-v2.yaml
-    # add label for Sysdig instance
+    if [ ! -z "$AGENT_SLIM" ]; then
+        DAEMONSET_FILE='/tmp/sysdig-agent-slim-daemonset-v2.yaml'
+    else
+        DAEMONSET_FILE='/tmp/sysdig-agent-daemonset-v2.yaml'
+    fi
+
     # -i.bak argument used for compatibility between mac (-i '') and linux (simply -i) 
+    sed -i.bak -e "s|# serviceAccount: sysdig-agent|serviceAccount: sysdig-agent|" $DAEMONSET_FILE
+
+    # Use IBM Cloud Container Registry instead of docker.io
+    sed -i.bak -e "s|\( *image: \)sysdig|\1icr.io/ext/sysdig|g" $DAEMONSET_FILE
+
+    # Pull the icr secrets if they don't exist
+    ICR_SECRET_EXIST=$(kubectl -n default get secret default-icr-io >/dev/null 2>&1 || echo 1)
+    if [ "$ICR_SECRET_EXIST" = 1 ]; then
+      # Throw an error instead of running the command for them because it could
+      #  take a long time for the secrets to become populated
+      echo "ERROR: default-icr-io secret doesn't exist in the default namespace"
+      echo "ERROR: Run: ibmcloud ks cluster pull-secret apply --cluster $IKS_CLUSTER_ID"
+      exit 1
+    fi
+    # Add the icr secret to our namespace. Delete beforehand to avoid conflicts
+    kubectl -n $NAMESPACE delete secret $NAMESPACE-icr-io 2>/dev/null || echo >/dev/null
+    kubectl get secret default-icr-io -n default -o yaml | sed "s/default/$NAMESPACE/" | kubectl apply -n $NAMESPACE -f -
+    # Use the pull secret in the daemonset flie. macOS's sed doesn't like \n
+    INDENT=$(grep 'containers' $DAEMONSET_FILE | sed 's/\( *\).*/\1/')
```

Comment:
`sed -i -e "s|\( *\)\(containers:\)|\1imagePullSecrets:\n\1- name: ibm-observe-icr-io\n\1\2|g" $DAEMONSET_FILE` doesn't work on macOS because the newline

Reply:
why not use `-o json` and modify the json with `jq`?

Reply:
I don't use `jq` much since users would have to install it. I see that it's already used in this script, I can refactor it to use `jq` if you prefer